### PR TITLE
Use Infura API v3

### PR DESCRIPTION
**Pull request by whymarrh**
This PR updates the project to use Infura's v3 API.

### CODE REVIEW

**Comment by whymarrh** [comment]
I've created #33 to note that rate limits aren't supported yet

**Comment by whymarrh** [comment]
I've created #34 to note that currently requests cannot be cancelled.

File: src/index.js [hunk 1]
```diff
@@ -77,7 +82,7 @@ async function performFetch (network, req, res, source) {
       case 405:
         throw rpcErrors.rpc.methodNotFound()
 
-      case 418:
```
Comment (whymarrh) [inline src/index.js:80]: I am not sure why this was here but it is no longer correct per the docs

File: test/index.js [hunk 2]
```diff
@@ -101,3 +124,41 @@ test('fetchConfigFromReq - source specified for request origin in header', (t) =
...
+
+test('createInfuraMiddleware throws with an invalid projectId', (t) => {
+
+  t.throws(() => createInfuraMiddleware({ projectId: 42 }), /Invalid value for 'projectId'/u)
+  t.throws(() => createInfuraMiddleware({ projectId: null }), /Invalid value for 'projectId'/u)
+  t.throws(() => createInfuraMiddleware({ projectId: undefined }), /Invalid value for 'projectId'/u)
+  t.throws(() => createInfuraMiddleware({ projectId: '' }), /Invalid value for 'projectId'/u)
+  t.end()
+
+})
```
Comment (rekmarks) [inline test/index.js:164]: It might be worth adding a test case for invalid headers but I won't block on it.

Comment (rekmarks) [inline test/index.js:164]: It might be worth adding a test case for invalid headers but I won't block on it.

**Review by rekmarks — APPROVED** [review]
LGTM! Just one optional suggestion.

File: test/index.js (hunk 2 above)
Comment (whymarrh) [reply test/index.js:164]: Updated with a few test cases

**Review by brad-decker — APPROVED** [review]

**Review by rekmarks — APPROVED** [review]

**Pull request merged**